### PR TITLE
Fix enhance-collection

### PR DESCRIPTION
**Pull request by thangngoc89**

### CODE REVIEW

File: src/enhance-collection/__tests__/index.js [hunk 1]
```diff
@@ -32,9 +40,63 @@ test("statinamic/lib/enhance-collection", (t) => {
...
+    [
+      {
+        k: "ey",
+        l: "hi",
+      }
+    ],
+  )
+})
+
+test("multiple filters", (t) => {
```
Comment (thangngoc89) [inline src/enhance-collection/__tests__/index.js]: @MoOx is this a valid test case for `filters` option ?

Comment (MoOx) [inline src/enhance-collection/__tests__/index.js]: yeah? What can be the issue?
We should also test that the array can be object or function too.

Comment (thangngoc89) [inline src/enhance-collection/__tests__/index.js]: The result is wrong.

Comment (thangngoc89) [inline src/enhance-collection/__tests__/index.js]: When I set filters: `["q", "k"]`, I suppose it will print any object that contains both `q` and `k` key

Comment (MoOx) [inline src/enhance-collection/__tests__/index.js]: filtering by an array is an && not and ||.

Comment (MoOx) [inline src/enhance-collection/__tests__/index.js]: result is the only object that contains both right?

Comment (thangngoc89) [inline src/enhance-collection/__tests__/index.js]: result

```
{ k: 'ey', l: 'hi' }
{ k: 'eye' }
```

Comment (thangngoc89) [inline src/enhance-collection/__tests__/index.js]: I do suppose it is `&&`

Comment (MoOx) [inline src/enhance-collection/__tests__/index.js]: Haha ok test is failing I missed that. So yes I guess we have to fix this according to the result you used.

Comment (thangngoc89) [inline src/enhance-collection/__tests__/index.js]: Yup. I'll fix that. Just ask this here to make sure I understand the code correctly :)

**Comment by MoOx** [comment]
Do you need a hand to fix this?

**Comment by thangngoc89** [comment]
Yes please. No idea where to get started. Feel free to use this branch as start point

**Comment by MoOx** [comment]
On it.

**Pull request closed without merging**